服务注册与发现容易panic挂掉

以下是其中一处:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4637eff]

goroutine 30 [running]:
github.com/bilibili/kratos/pkg/naming/discovery.(*Discovery).polls(0xc0002d2c80, 0x49c2120, 0xc0003107c0, 0x2, 0x0, 0x0)
       /go/src/bilibili/kratos/pkg/naming/discovery/discovery.go:598 +0x9df
github.com/bilibili/kratos/pkg/naming/discovery.(*Discovery).serverproc(0xc0002d2c80)
       /go/src/bilibili/kratos/pkg/naming/discovery/discovery.go:518 +0x16f
created by github.com/bilibili/kratos/pkg/naming/discovery.(*Discovery).Build.func1
        /go/src/bilibili/kratos/pkg/naming/discovery/discovery.go:250 +0x3e
exit status 2

--- a/pkg/naming/discovery/discovery.go
+++ b/pkg/naming/discovery/discovery.go
@@ -609,6 +609,10 @@ func (d *Discovery) polls(ctx context.Context) (apps map[string]*naming.Instance
 func (d *Discovery) broadcast(apps map[string]*naming.InstancesInfo) {
 	for appID, v := range apps {
 		var count int
+		// v maybe nil in old version(less than v1.1) discovery,check incase of panic
+		if v==nil {
+			continue
+		}
 		for zone, ins := range v.Instances {
 			if len(ins) == 0 {
 				delete(v.Instances, zone)

